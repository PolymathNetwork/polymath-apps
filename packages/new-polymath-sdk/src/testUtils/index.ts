import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
