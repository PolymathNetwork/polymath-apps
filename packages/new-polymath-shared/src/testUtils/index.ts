export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
