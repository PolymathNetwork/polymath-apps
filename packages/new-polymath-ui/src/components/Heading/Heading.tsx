import React, { FC } from 'react';
import styled from 'styled-components';
import {
  variant,
  color,
  ColorProps,
  fontWeight,
  FontWeightProps,
  lineHeight,
  LineHeightProps,
  fontSize,
  FontSizeProps,
} from 'styled-system';

import { Box } from '~/components/Box';

const headingStyle = variant({
  key: 'headings',
});

export type HeadingProps = {
  variant: 'h1' | 'h2' | 'h3' | 'h4';
} & ColorProps &
  FontWeightProps &
  FontSizeProps &
  LineHeightProps;

const StyledHeading = styled(Box)<HeadingProps>`
  ${headingStyle};
  ${color};
  ${fontWeight};
  ${lineHeight};
  ${fontSize};
`;

// TODO @grsmto: remove when https://github.com/pedronauck/docz/issues/337 is resolved
export const HeadingDocz: FC<HeadingProps> = (props: any) => {
  return <Heading {...props} />;
};

export const Heading = Object.assign(StyledHeading, {
  defaultProps: {
    as: 'h2',
    variant: 'h3',
    lineHeight: 'normal',
    mt: 0,
    mb: 'm',
  },
});
