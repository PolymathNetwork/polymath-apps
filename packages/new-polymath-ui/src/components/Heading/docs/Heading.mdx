---
name: Heading
menu: Primitives
---

import { Playground, PropsTable } from 'docz';

import { Heading, HeadingDocz } from '../';

# Heading

<PropsTable of={HeadingDocz} />

## Basic usage

<Playground>
  <Heading>This is a Heading.</Heading>
</Playground>
