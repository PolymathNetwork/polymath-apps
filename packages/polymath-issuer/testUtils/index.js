import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import * as factories from './factories';

const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);

export { factories };
