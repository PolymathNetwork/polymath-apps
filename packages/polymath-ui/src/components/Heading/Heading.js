import React from 'react';
import styled from 'styled-components';
import { variant } from 'styled-system';

import Paragraph from '../Paragraph';

const headingStyle = variant({
  key: 'headings',
});

const Heading = styled(Paragraph)`
  ${headingStyle};
`;

export default Heading;

Heading.defaultProps = {
  variant: 'h3',
  as: 'h2',
  lineHeight: 'normal',
  fontWeight: 'bold',
  mb: 'm',
};
