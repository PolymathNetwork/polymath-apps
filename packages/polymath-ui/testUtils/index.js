import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
