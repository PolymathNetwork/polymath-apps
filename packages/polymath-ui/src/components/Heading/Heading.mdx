---
name: Heading
menu: Primitives
---

import { Playground, PropsTable } from 'docz';

import Heading from './Heading';

# Heading

<PropsTable of={Heading} />

## Basic usage

<Playground>
  <Heading>This is a Heading.</Heading>
</Playground>
