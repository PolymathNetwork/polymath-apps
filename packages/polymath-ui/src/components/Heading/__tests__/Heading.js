import React from 'react';
import { render } from '../../../../testUtils';

import Heading from '../';

test('renders without crashing', () => {
  const { container } = render(<Heading>This is a Heading.</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});
