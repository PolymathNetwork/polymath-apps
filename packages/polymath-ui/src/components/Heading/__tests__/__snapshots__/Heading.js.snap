// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<h2
  class="Paragraph-sc-1ehtinw-0 Heading-sc-10pyzom-0 hVpnIj"
>
  This is a Heading.
</h2>
`;
